sql/pre_install/tables.sql

```diff
@@ -333,16 +333,22 @@ SELECT pg_catalog.pg_extension_config_dump('_timescaledb_catalog.continuous_agg'
 -- See the comments for ContinuousAggsBucketFunction structure.
 CREATE TABLE _timescaledb_catalog.continuous_aggs_bucket_function (
   mat_hypertable_id integer NOT NULL,
-  -- The schema of the function. Equals TRUE for "timescaledb_experimental", FALSE otherwise.
-  experimental bool NOT NULL,
+  -- The schema of the function.
+  bucket_func_schema name NOT NULL,
   -- Name of the bucketing function, e.g. "time_bucket" or "time_bucket_ng"
-  name text NOT NULL,
+  bucket_func_name name NOT NULL,
```

Comment:
Why not use [`REGPROCEDURE`](https://www.postgresql.org/docs/current/datatype-oid.html#DATATYPE-OID-TABLE) insted of name for the bucket function?

Reply:
I thought about it. However, I wanted to be compatible with our other approaches. See:

* https://github.com/timescale/timescaledb/blob/d93aa5c8c1736874ffe647f308495e18b904e086/sql/pre_install/tables.sql#L101-L102
* https://github.com/timescale/timescaledb/blob/d93aa5c8c1736874ffe647f308495e18b904e086/sql/pre_install/tables.sql#L229-L230

Reply:
Does this simplify interfacing with these other places? If it's only for visual similarity, I'd say we should use `regprocedure` which is a more "proper" approach (has more convenient functions to work with it, properly survives dump/restore and so on).

Reply:
Isn't regprocedure an Oid? Oids can differ based on the instance. This is the reason we don't use Oids in our catalog tables. I am pretty sure we have a doc describing this. cc @mkindahl

Reply:
@fabriziomello Mentioned that PostgreSQL might do some magic for `regprocedure` (e.g., during a dump and restore) and adjust the value automatically. I am currently performing a test if this works as expected.

Reply:
Does that work with binary dumps too?

Reply:
Yes it works @gayyappan ... and we already have it in our catalog: https://github.com/timescale/timescaledb/blob/main/sql/pre_install/tables.sql#L414

Reply:
@gayyappan I tried this also with dumps in tar format (`pg_dump ... -F t ..`). It works as well. Does this answer your question, or did you have another format in mind?

Reply:
> Isn't regprocedure an Oid? Oids can differ based on the instance. This is the reason we don't use Oids in our catalog tables. I am pretty sure we have a doc describing this. cc @mkindahl

Typically OIDs using a "regxyz" type are dumped in text format so it should work well even when moving between instances. The problematic case is when using a generic OID, which is treated just as a number, and not a "regxyz" type. It is described in [Using Object Identifier Types for Catalog](https://docs.google.com/document/d/1boLas5-4BojZStnnecR5drrm6zyI8R8kerVj_nuVSOc/edit?usp=sharing).